### PR TITLE
refactor(knowledge-layer): Knowledge Layer improvements

### PR DESCRIPTION
**Pull request by peterychang**
#### Overview

Every knowledge layer subsystem had functionally identical implementations for summary generation. Unified all paths to a single implementation.

Additionally register file names to summary store even when not generating summaries. Experimentation shows this can improve query results.

#### Validation

- [x] I ran the relevant local checks or explained why they are not applicable.
- [x] I added or updated tests for behavior changes.
- [x] I updated documentation for user-facing or contributor-facing changes.
- [x] I confirmed this PR does not include secrets, credentials, or internal-only data.
- [x] I certify this contribution under the Developer Certificate of Origin (DCO) and signed my commits with `git commit -s` or an equivalent sign-off.
- [x] I replaced the DCO sign-off placeholder with my GitHub commit identity and kept the required angle brackets around the email address.

## Summary by CodeRabbit

* **New Features**
  * Added consistent document summaries across supported search and retrieval integrations.
  * Summaries use concise sentence formatting and configurable input limits.
  * Files without generated summaries now receive a “No summary available” placeholder.
* **Bug Fixes**
  * Summary-generation failures no longer interrupt ingestion.
  * Summary records are consistently cleaned up when files or collections are deleted.
  * Improved handling when documents, language models, or summaries are unavailable.
  * Standardized summary cleanup and output truncation.

### CODE REVIEW

File: sources/knowledge_layer/src/utils.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    try:
+        response = llm.invoke(prompt)
+        summary = getattr(response, "content", None) or str(response)
+        if not summary:
+            return None
+        summary = summary.strip()
+        logger.debug(f"Summary generated for {file_name}", stacklevel=2)
+        return summary
+    except Exception as e:
+        logger.warning(f"Summary via LLM failed for {file_name}: {e}", stacklevel=2)
```
[the remark this answers is not in the document]
Comment (peterychang) [reply sources/knowledge_layer/src/utils.py]: Leaving filename logging behavior the same for each subsystem

File: sources/knowledge_layer/src/llamaindex/adapter.py [hunk 2]
```diff
@@ -1592,7 +1562,9 @@ def generate_summary(self, text_content: str, file_name: str) -> str | None:
         """Generate summary using NVIDIA NIM if enabled."""
         if not self.generate_summary_enabled:
             return None
-        return _generate_document_summary(text_content, file_name, self.summary_llm)
+        return summarize_document(
+            text_content, self.summary_llm, file_name=file_name, input_max_chars=SUMMARY_MAX_INPUT_CHARS
+        )
```
[the remark this answers is not in the document]
Comment (peterychang) [reply sources/knowledge_layer/src/llamaindex/adapter.py]: pre-change behavior is maintained

File: sources/knowledge_layer/src/utils.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        "Summarize this uploaded document in one concise sentence for a research assistant. "
+        "Focus on the document's topic and likely usefulness.\n\n"
+        f"Content Excerpt:\n{text}"
+    )
+    try:
+        response = llm.invoke(prompt)
+        summary = getattr(response, "content", None) or str(response)
+        summary = summary.strip() if summary else None
+        if not summary:
+            return None
```
Comment (tanleach) [inline sources/knowledge_layer/src/utils.py:50]: Could we distinguish missing `response.content` from content that is present but empty? With the current `or str(response)` fallback, `AIMessage(content="")` becomes the message’s metadata representation and may be stored as the document summary.

This preserves the fallback for response types without content while treating exact-empty content as unavailable:

```suggestion
        content = getattr(response, "content", None)
        summary = str(response) if content is None else content
        summary = summary.strip() if summary else None
        if not summary:
            return None
```

It would also be helpful to make the existing regression test use `SimpleNamespace(content="")` so the exact-empty case remains covered.

**Review by tanleach — COMMENTED** [review]
One actionable suggestion for the exact-empty LLM response path.

File: sources/knowledge_layer/src/foundational_rag/adapter.py [hunk 4]
```diff
@@ -860,18 +862,17 @@ def _upload_files_async(
...
-                except Exception as e:
-                    logger.debug(f"Summary generation failed for {file_name}: {e}")
+        # Collect summaries from parallel generation if enabled and register them
+        # otherwise use the default summary
+        # TODO: The summary is registered BEFORE the file ingestion is checked. This can result in
+        #       a state where the summary is registered but the file is either unavailable or failed.
+        for file_name, future in summary_futures:
+            try:
+                summary = future.result(timeout=30) if future else _DEFAULT_SUMMARY
+                register_summary(collection_name, file_name, summary)
```
Comment (tanleach) [inline sources/knowledge_layer/src/foundational_rag/adapter.py:872]: should upload acceptance depend on a successful summary-store update? Both this batch path and `upload_file()` continue into summary registration after `response.raise_for_status()` fails. Because summaries are upserted by `(collection, filename)`, a rejected duplicate-name upload can overwrite an existing document’s summary.

**Review by tanleach — COMMENTED** [review]
One question about the Foundational RAG upload/summary-store lifecycle.
